### PR TITLE
Add SystemSessionPropertiesProvider

### PR DESCRIPTION
**Pull request by lukasz-stec**
Add option to configure many SystemSessionPropertiesProviders
in the SessionPropertyManager

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/dispatcher/DispatchManager.java [hunk 1]
```diff
@@ -89,6 +90,7 @@ public DispatchManager(
             AccessControl accessControl,
             SessionSupplier sessionSupplier,
             SessionPropertyDefaults sessionPropertyDefaults,
+            SessionPropertyManager sessionPropertyManager,
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/dispatcher/DispatchManager.java:93]: nit: injecting `SessionPropertyManager` can be a separate preparatory commit.

File: core/trino-main/src/main/java/io/trino/SystemSessionPropertiesProvider.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ */
+package io.trino;
+
+import io.trino.spi.session.PropertyMetadata;
+
+import java.util.List;
+
+public interface SystemSessionPropertiesProvider
+{
+    List<PropertyMetadata<?>> getSessionProperties();
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/SystemSessionPropertiesProvider.java:22]: Set?

Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/SystemSessionPropertiesProvider.java:22]: OH - it was list before. Keep it.

**Review by losipiuk — APPROVED** [review]
LGTM

File: core/trino-main/src/main/java/io/trino/dispatcher/DispatchManager.java (hunk 1 above)
Comment (lukasz-stec) [reply core/trino-main/src/main/java/io/trino/dispatcher/DispatchManager.java:93]: done

**Pull request merged**